drivers/blink/CMakeLists.txt

```diff
@@ -2,4 +2,9 @@
 # SPDX-License-Identifier: Apache-2.0
 
 zephyr_library()
+zephyr_iterable_section(
+  NAME blink_driver_api
+  KVMA RAM_REGION GROUP RODATA_REGION
+  SUBALIGN CONFIG_LINKER_ITERABLE_SUBALIGN)
+
```

Comment:
looks like this is too much to create a new driver class. can't this be wrapped?

Reply:
I could add something like `zephyr_api_section`? CMake iterable sections are used a lot too in the zephyr tree though.

Reply:
+1 on the wrapper. There will be such addition for all device APIs. In the end it could reduce the amount of added lines quite a bit.

Reply:
Sure, what needs to be configurable?

```cmake
macro(zephyr_api_section class align)
  zephyr_iterable_section(
    NAME "${class}_driver_api"
    KVMA RAM_REGION GROUP RODATA_REGION
    SUBALIGN ${align}
  )
endmacro()

# ...

zephyr_api_section(adc CONFIG_LINKER_ITERABLE_SUBALIGN)
```

Are the regions/groups something we should be hardcoding? Where do we stop abstracting?

Reply:
From a user perspective, we're just adding a new device driver class, so I'd expect users to specify only the driver class name, eg `zephyr_device_class(blink)`.

Reply:
I'm looking into generating it at build time for `CONFIG_CMAKE_LINKER_GENERATOR` similar to LD templates, so this won't be needed.